transformer_engine/jax/cpp_extensions/attention.py

```diff
@@ -2617,10 +2737,32 @@ def fused_attn_fwd(
         assert bias is None
         bias = jnp.zeros(0, dtype=qkv[0].dtype)
 
+    if softmax_offset is None:
+        assert softmax_type != AttnSoftmaxType.LEARNABLE_SOFTMAX, f"Unknown {softmax_type=}"
```

Comment:
I think "Unknown AttnSoftmaxType.LEARNABLE_SOFTMAX" might not be the right assert to raise as it is a known type, if I understand this right. You are just trying to ensure that if it is learnable then there's always a softmax_offset to accompany it, right ?

```suggestion
        assert softmax_type != AttnSoftmaxType.LEARNABLE_SOFTMAX, f"{softmax_type=} must have softmax_offset"
```